### ISSUE TITLE
Exceptions are not explicit enough

### ISSUE THREAD

**Issue by wsw70** [custom-components/pyscript#107]
I wrote yesterday a simple automation for three rooms to have their shutters move up and lights switched off (both `cover.<someone>` and `group.<someone>` exist (the latter being a set of switches that manage lights))

```
@time_trigger("once(09:45)")
def volets_matin_total():
    for who in ['martin', 'michael', 'parents']:
        cover.open_cover(entity_id=f"cover.{who}_volets")
        homeassistant.turn_off(f"group.{who}")
```
I got this morning  the follwing error (first time it actually ran - I do not think there is a way to test a `pyscript `automation the same way as in the GUI where there is the 'Execute' function)

```
2020-12-03 09:45:00 ERROR (MainThread) [custom_components.pyscript.file.volets.volets_matin_total] Exception in <file.volets.volets_matin_total> line 15:
            homeassistant.turn_off(f"group.{who}")
                                            ^
TypeError: exceptions must derive from BaseException
```

So something bad happened (not yet sure what) but the traceback is not very talkative because it seems that this is rather an information about the nature of the raised Exception not being correct rather than the traceback of the error in my code itself. This is not the first time I see that.

Or maybe I misunderstood something?

**Comment by craigbarratt** [comment]
I don't see a problem with your code.  This might be a bug in pyscript - it seems that somewhere an invalid exception is being raised.

If you are willing to edit the code for pyscript, here's how to get more traceback information. In `custom_components/pyscript/eval.py` add
```
import traceback
```
near the top of the file, and at the end of the `format_exc()` function ([L1936 for 1.0.0](https://github.com/custom-components/pyscript/blob/59283b8f2eb799083801f282a14274c04e17bf35/custom_components/pyscript/eval.py#L1936)), replace:
```
return mesg
```
with
```
return mesg + "\n" + traceback.format_exc(10)
```
You should get a more detailed stack trace.

**Comment by craigbarratt** [comment]
Ah - for service calls you need to specify all the parameters as keyword parameters, not as positional arguments.

**Comment by wsw70** [comment]
@craigbarratt  Thank you for your answer

> Ah - for service calls you need to specify all the parameters as keyword parameters, not as positional arguments.

Great, I changed that and will see tomorrow if it works. However ...

> BTW, a lot of interactive testing and development can be done using Jupyter. Have you tried that yet?

I used Jupyter a few times over the last years but never really tried to use it for something substantial. Thi sis THE right opportunity to learn it. Thanks for the suggestion, I will try today (already installed it and started the tutorial but jumped to standard IDE-based coding afterward)
